packages/invoices/src/OutboundShipment/DetailView/modals/ItemDetailsModal.tsx

```diff
@@ -169,44 +181,64 @@ export const ItemDetailsModal: React.FC<ItemDetailsModalProps> = ({
 
   const checkAllocatedQuantities = () => {
     const values = getValues();
-    const allocatedQuantity = batchRows.reduce(
-      (total, batch) => (total += Number(values[batch.id] || 0)),
+    const allocatedUnits = batchRows.reduce(
+      (total, batch) =>
+        (total += Number(values[batch.id] || 0) * batch.packSize),
       Number(values['placeholder'] || 0)
     );
-
-    setIsAllocated(allocatedQuantity >= quantity && allocatedQuantity > 0);
+    setAllocated(Math.floor(allocatedUnits / packSize));
   };
 
   const allocateQuantities = () => {
     // if invalid quantity entered, don't allocate
-    if (quantity < 1 || Number.isNaN(quantity)) {
+    if (quantity < 1 || Number.isNaN(quantity) || !selectedItem) {
       return;
     }
     // if the selected item has no batch rows, allocate all to the placeholder
     if (batchRows.length === 0) {
       setValue('placeholder', quantity);
-      setIsAllocated(true);
+      setAllocated(quantity);
       return;
     }
 
-    let toAllocate = 0;
-    toAllocate += quantity;
+    // calculations are normalised to units
+    let toAllocate = quantity * packSize;
+    let batchAllocation = 0;
 
     batchRows.forEach(batch => {
-      const allocatedQuantity = Math.min(
-        toAllocate,
-        batch.availableNumberOfPacks * batch.packSize
-      );
-      toAllocate -= allocatedQuantity;
-      setValue(batch.id, allocatedQuantity);
+      batchAllocation = 0;
```

Comment:
the addition of pack size selection has made this bigger than I wanted. I _think_ it is all worth it?
we do have a lot of trouble with quantity / pack size confusion

Reply:
Hopefully 😁  it is becoming quite a beast!